src/server/json_family.cc

```diff
@@ -2045,130 +1730,97 @@ void JsonFamily::NumMultBy(CmdArgList args, ConnectionContext* cntx) {
     return;
   }
 
-  JsonPathV2 expression = PARSE_PATHV2(path);
+  WrappedJsonPath json_path = GET_OR_SEND_UNEXPECTED(ParseJsonPath(path));
 
   auto cb = [&](Transaction* t, EngineShard* shard) {
-    return OpDoubleArithmetic(t->GetOpArgs(shard), key, path, dnum, OP_MULTIPLY,
-                              std::move(expression));
+    return OpDoubleArithmetic(t->GetOpArgs(shard), key, json_path, dnum, OP_MULTIPLY);
   };
 
-  auto* rb = static_cast<RedisReplyBuilder*>(cntx->reply_builder());
   OpResult<string> result = cntx->transaction->ScheduleSingleHopT(std::move(cb));
-
-  if (result) {
-    rb->SendBulkString(*result);
-  } else {
-    cntx->SendError(result.status());
-  }
+  auto* rb = static_cast<RedisReplyBuilder*>(cntx->reply_builder());
+  reply_generic::Send(result, rb);
 }
 
 void JsonFamily::Toggle(CmdArgList args, ConnectionContext* cntx) {
-  string_view key = ArgS(args, 0);
-  string_view path = ArgS(args, 1);
-
-  JsonPathV2 expression = PARSE_PATHV2(path);
-
-  auto cb = [&](Transaction* t, EngineShard* shard) {
-    return OpToggle(t->GetOpArgs(shard), key, path, std::move(expression));
-  };
+  CmdArgParser parser{args};
+  string_view key = parser.Next();
+  string_view path = parser.NextOrDefault();
 
-  Transaction* trans = cntx->transaction;
-  OpResult<vector<OptBool>> result = trans->ScheduleSingleHopT(std::move(cb));
+  WrappedJsonPath json_path = GET_OR_SEND_UNEXPECTED(ParseJsonPath(path));
 
-  if (result) {
-    PrintOptVec(cntx, result);
+  if (json_path.IsLegacyModePath()) {
+    OpToggle<bool>(key, json_path, cntx);
   } else {
-    cntx->SendError(result.status());
+    OpToggle<long>(key, json_path, cntx);
   }
 }
```

Comment:
we usually have Op prefix reserved for shard operations, might confuse someone. If you're using it just as a type transfer function, you can do

`auto op = legacy ? OpToggle<bool> : OpToggle<long>`

and then all the transaction + reply part

Note that OpToggles return type can be changed from auto to void

Reply:
Rename to `EvaluateToggle`